packages/client/lib/sync/fullsync.ts

```diff
@@ -81,6 +80,10 @@ export class FullSynchronizer extends Synchronizer {
         hash
       )} hardfork=${this.config.chainCommon.hardfork()}`
     )
+    if (this.config.mine) {
+      // Start the TxPool immediately if mining
+      this.txPool.start()
```

Comment:
I am a bit confused by this distinction. Shouldn't the miner as well only start once sync of an eventual existing chain is completed?

Reply:
Well if this is a private dev network scenario where you are the only miner and connect to no peers, a sync will never "start" to "complete" to then start the tx pool. It will just keep searching for potential peers. That was the scenario I was facing which is why I added this here.